src/context/EntryModalContext.tsx

```diff
@@ -0,0 +1,62 @@
+'use client';
+
+import { createContext, useContext, useMemo, useState } from 'react';
+
+import { ChildrenFC } from '../utils/type';
+
+type ModalContextProps = {
+  isEntryModalOpen: boolean;
+  openEntryModal: () => void;
+  closeEntryModal: () => void;
+  onSignedUp: (() => void) | undefined;
+  onSignedUpComplete: (callback: () => void) => void;
+};
+
+const ModalContext = createContext<ModalContextProps | null>(null);
```

Comment:
Definitely, a good job, you made the context work. It is always quite tricky to wrap a head around. We need to move the structure/flow of data somehow around so it does not feel so all over the place. I will have a look later